Add Microsoft.Web ResourceHealthMetadata swagger

pavelzel opened the pull request:
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

pavelzel committed:
Adding Microsoft.Web ResourceHealthMetadata APIs

Adding a new set of Resource Health Metadata API swagger files for Microsoft.Web

pavelzel committed:
Update autorest config to test Microsoft.Web ResourceHealthMetadata swagger file

Update the Microsoft.Web autorest config to test validity of the ResourceHealthMetadata swagger file

AutorestCI commented:
# Automation for azure-sdk-for-python

AutorestCI commented:
# Automation for azure-sdk-for-go

fearthecowboy commented:
Greetings @pavelzel !

These look like new APIs, so I'm going to ask @ravbhatnagar  to do a quick review as well. 

Straight off they look pretty good; were you going to add `x-ms-examples` for these APIs?

pavelzel committed:
Adding ResourceHealthMetadata examples

Adding examples for the Microsoft.Web ResourceHealthMetdata APIs

AutorestCI commented:
# Automation for azure-sdk-for-node

pavelzel commented:
@fearthecowboy Added x-ms-examples as requested

pavelzel commented:
@fearthecowboy @ravbhatnagar , is there anything else you need from me for this PR?

fearthecowboy commented:
Your examples are failing validation: https://travis-ci.org/Azure/azure-rest-api-specs/jobs/350123757

ravbhatnagar commented on specification/web/resource-manager/Microsoft.Web/stable/2016-03-01/ResourceHealthMetadata.json:
```diff
@@ -0,0 +1,373 @@
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
```
Add "default" response to all APIs to capture error cases. Schema complies with the ARM RPC error contract. example from Batch RP swagger.

pavelzel replied:
@ravbhatnagar This looks to be a newer requirement for swagger files. Since we (Microsoft.Web) auto-generate our swagger files from code, is it okay to not have the "default" response for this file while we work on adding support for this in the auto-generation?

pavelzel replied:
Fixed up the responses to have a default error response

ravbhatnagar commented on specification/web/resource-manager/Microsoft.Web/stable/2016-03-01/ResourceHealthMetadata.json:
```diff
@@ -0,0 +1,373 @@
+        "properties": {
+          "description": "ResourceHealthMetadata resource specific properties",
+          "properties": {
+            "category": {
```
Is this a candidate for an enum?

pavelzel replied:
This is a string as per spec provided by the RHC team, relevant part of spec:
Category (string) - this is the category of the sku that you have communicated to RHC. RHC Rp will select the policy based on this value. If the sku's category does not match existing categories, we will fall back to the default category

ravbhatnagar commented on specification/web/resource-manager/Microsoft.Web/stable/2016-03-01/ResourceHealthMetadata.json:
```diff
@@ -0,0 +1,373 @@
+              "description": "The category that the resource matches in the RHC Policy File",
+              "type": "string"
+            },
+            "signalAvailability": {
```
ideally this should be an enum with multiple states. Currently you might just require true or false for this but in future as scenario evolves you might want to indicate more states.

pavelzel replied:
Same as category, this is a boolean as per the spec I was given by RHC team:
SignalAvailability (boolean) - This field is to notify us if the signal for a resource is reaching us or not. This can come into picture when you are aware that a particular set of resources (possibly belonging a test region or a legacy region) are not sending RHC signals. When this is false, we will put the corresponding resource into unknown events bucket.

ravbhatnagar commented:
@fearthecowboy  - no blocking feedback from my side. Feel free to merge based on response to above comments.

pavelzel committed:
Correcting examples for Microsoft.Web ResourceHealthMetadata APIs

The location should not have been populated for the Microsoft.Web ResourceHealthMetadata API examples. The single ResourceHealthMetadata GET call should return ResourceHealthMetadata and not a collection of them

AutorestCI commented:
# Automation for azure-libraries-for-java

pavelzel committed:
Microsoft.Web ResourceHealthMetadata API adding default error response

Adding the default errror responses to the Microsoft.Web ResourceHealthMetadata API swagger

pavelzel committed:
Renaming ErrorResponse to DefaultErrorResponse for default errors

For Microsoft.Web ResourceHealthMetadata APIs renaming the default error response from ErrorResponse to DefaultErrorResponse to fix the python SDK generation. The DefaultErrorResponse will be the default error response for our other APIs in the future as well

jhendrixMSFT commented:
@AutorestCI rebuild azure-sdk-for-go

pavelzel commented:
@fearthecowboy is there anything else you need from me for this PR?

fearthecowboy commented:
Looks good to me @pavelzel  -- you want me to merge this now?

pavelzel commented:
@fearthecowboy Yes, please go ahead and merge. Thank you!

jhendrixMSFT commented:
@AutorestCI rebuild azure-sdk-for-go
